Editorial review: Document interest invokers (#42021)
mdn/content: files/en-us/web/api/htmlelement/interest_event/index.md

---
title: "HTMLElement: interest event"
slug: Web/API/HTMLElement/interest_event
page-type: web-api-event
browser-compat: api.HTMLElement.interest_event
---

{{APIRef("HTML DOM")}}

> The **`interest`** event of the {{domxref("HTMLElement")}} interface is fired on an [interest invoker](/en-US/docs/Web/API/Popover_API/Using_interest_invokers)'s target element when interest is shown, allowing code to be run in response.

Reviewer: ```suggestion
The **`interest`** event of the {{domxref("HTMLElement")}} interface is fired on the target element of an [interest invoker](/en-US/docs/Web/API/Popover_API/Using_interest_invokers) when interest is shown, allowing code to be run in response.
```

Reviewer: Ah, got it

## Syntax

Use the event name in methods like {{domxref("EventTarget.addEventListener", "addEventListener()")}}, or set an event handler property.

```js-nolint
addEventListener("interest", (event) => { })

oninterest = (event) => { }
```

## Event type

An {{domxref("InterestEvent")}}. Inherits from {{domxref("Event")}}.

{{InheritanceDiagram("InterestEvent")}}

## Examples

### Basic interest invoker event usage

In this example, we use the `interest` and `loseinterest` events to report when interest is being shown and lost on a {{htmlelement("button")}} element interest invoker. We do this by printing messages into the {{htmlelement("p")}} target element's text content.

#### HTML

> We set up a relationship between the `<button>` element interest invoker and its target `<p>` element by setting the `<button>` element's `interestfor` attribute equal to the `<p>` element's `id`.

Reviewer: ```suggestion
We set up the relationship between the `<button>` element interest invoker and its target `<p>` element by setting the value of the `<button>` element's `interestfor` attribute equal to the `<p>` element's `id`.
```

```html live-sample___basic-interest-invoker
<button href="#" interestfor="mytarget">Interest invoker</button>

> <p id="mytarget">Interest not being shown.</p>

Reviewer: ```suggestion
<p id="mytarget">No interest being shown currently.</p>
```

```

#### JavaScript

We grab a reference to the `<button>` element and its target element (via its {{domxref("HTMLButtonElement.interestForElement", "interestForElement")}} property).

```js live-sample___basic-interest-invoker
const invoker = document.querySelector("[interestfor]");
const target = invoker.interestForElement;
```

We then set two event listeners on the target element, for the `interest` and `loseinterest` events.

- When interest is shown, we print a message into the target `<p>` element's text context to say that interest has been shown via the `<button>` element. Note how you can get a reference to the interest invoker via the event object's {{domxref("InterestEvent.source", "source")}} property.

> - When interest is lost, we report that, again via a message printed to the paragraph text.

Author: I added this change manually, as I didn't trust the diff on here — it looked like it might result in a duplicate paragraph if I accepted it.

```js live-sample___basic-interest-invoker
target.addEventListener("interest", (e) => {

>   target.textContent = `Interest being shown via the ${e.source.tagName} element.`;

Reviewer: ```suggestion
  target.textContent = `Interest shown via the ${e.source.tagName} element.`;
```

---
title: "HTMLElement: interest event"
slug: Web/API/HTMLElement/interest_event
page-type: web-api-event
browser-compat: api.HTMLElement.interest_event
---

{{APIRef("HTML DOM")}}

The **`interest`** event of the {{domxref("HTMLElement")}} interface is fired on an [interest invoker](/en-US/docs/Web/API/Popover_API/Interest_invokers)'s target element when interest is shown, allowing code to be run in response.

> This event is not [cancelable](/en-US/docs/Web/API/Event/cancelable).

Author: Ah, gotcha. I've updated both pages as appropriate.
